crates/shared/src/order_validation.rs

```diff
@@ -1647,70 +1582,6 @@ mod tests {
         assert!(matches!(result, Err(ValidationError::ZeroAmount)));
     }
 
-    #[tokio::test]
-    async fn post_out_of_market_orders_when_limit_orders_disabled() {
-        let expected_buy_amount = U256::from(100);
-
-        let mut order_quoter = MockOrderQuoting::new();
-        let mut bad_token_detector = MockBadTokenDetecting::new();
-        let mut balance_fetcher = MockBalanceFetching::new();
-        order_quoter.expect_find_quote().returning(move |_, _| {
-            Ok(Quote {
-                buy_amount: expected_buy_amount,
-                sell_amount: U256::from(1),
-                fee_amount: U256::from(1),
-                ..Default::default()
-            })
-        });
-        bad_token_detector
-            .expect_detect()
-            .returning(|_| Ok(TokenQuality::Good));
-        balance_fetcher
-            .expect_can_transfer()
-            .returning(|_, _| Ok(()));
-        let mut limit_order_counter = MockLimitOrderCounting::new();
-        limit_order_counter.expect_count().returning(|_| Ok(0u64));
-        let validator = OrderValidator::new(
-            dummy_contract!(WETH9, [0xef; 20]),
-            Arc::new(order_validation::banned::Users::none()),
-            OrderValidPeriodConfiguration::any(),
-            false,
-            Arc::new(bad_token_detector),
-            dummy_contract!(HooksTrampoline, [0xcf; 20]),
-            Arc::new(order_quoter),
-            Arc::new(balance_fetcher),
-            Arc::new(MockSignatureValidating::new()),
-            Arc::new(limit_order_counter),
-            0,
```

Comment:
@MartinquaXD I investigated this test, the issue is that the limit for user limit orders is 0, and the order it is trying to place is a limit order because it is not "in-market", so the test fails. I thought this test is superfluous since we have these two tests:
- https://github.com/cowprotocol/services/pull/2496/files#diff-1ff024ca2dfa09435b86bbecf6919e50319d070f04f71ed834341d5d254b3c15L1451
- https://github.com/cowprotocol/services/pull/2496/files#diff-1ff024ca2dfa09435b86bbecf6919e50319d070f04f71ed834341d5d254b3c15L1532